### PR TITLE
upgrade dbt-bigquery and re-generate artifact types

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description
_Describe your changes and why you're making them. Please include the context, motivation, and relevant dependencies._

Resolves https://github.com/cal-itp/data-infra/issues/2546
Pre-requisite for https://github.com/cal-itp/data-infra/issues/2540 (need dbt 1.5)

Upgrades dbt to 1.5 and begins generating the artifact Pydantic models from the publish JSON Schema definitions.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

## How has this been tested?
_Include commands/logs/screenshots as relevant._

Visualize: tested locally (CI still disabled)
Publish: docker-compose run
Run/upload: docker-compose run

## Post-merge follow-ups
_Document any actions that must be taken post-merge to deploy or otherwise implement the changes in this PR (for example, running a full refresh of some incremental model in dbt). If these actions will take more than a few hours after the merge or if they will be completed by someone other than the PR author, please create a dedicated follow-up issue and link it here to track resolution._

- [x] No action required
- [ ] Actions required (specified below)

### CODE REVIEW

**Review by SorenSpicknall — APPROVED** [review]

File: warehouse/scripts/dbt_artifacts/__init__.py [hunk 1]
```diff
@@ -33,126 +47,58 @@ def get_engine(project, max_bytes=None):
...
+    database: Optional[str]
+    columns: Optional[Dict[str, ColumnInfo]] = {}
 
     def __init__(self, **kwargs):
-        super(BaseNode, self).__init__(**kwargs)
+        super(NodeModelMixin, self).__init__(**kwargs)
         self._instances[self.unique_id] = self
-        for column in self.columns.values():
-            column.parent = self
+        # for column in self.columns.values():
```
Comment (lauriemerrell) [inline warehouse/scripts/dbt_artifacts/__init__.py:92]: wondering why this is commented?

File: warehouse/scripts/dbt_artifacts/__init__.py [hunk 2]
```diff
@@ -193,119 +139,159 @@ def gvattrs(self) -> Dict[str, Any]:
...
+
+class TileFormat(str, Enum):
+    mbtiles = "mbtiles"
+    pbf = "pbf"
 
 
-class Owner(BaseModel):
-    name: Optional[str]
-    email: str
+# class Column(BaseColumn):
```
Comment (lauriemerrell) [inline warehouse/scripts/dbt_artifacts/__init__.py:256]: what is all this commented content?

File: warehouse/scripts/dbt_artifacts/manifest.py [hunk 3]
```diff
@@ -0,0 +1,1438 @@
...
+
+
+class MacroDependsOn(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    macros: Optional[List[str]] = []
+
+
+class ResourceType9(Enum):
```
Comment (lauriemerrell) [inline warehouse/scripts/dbt_artifacts/manifest.py:291]: I find it confusing that the `ResourceType`s are just defined sporadically throughout

**Review by lauriemerrell — APPROVED** [review]
Wondering if you can clean up some of the comments. Also I don't love how datamodel-codegen handles some of the enums but I guess there's nothing really to be done

File: warehouse/scripts/dbt_artifacts/manifest.py (hunk 3 above)
Comment (atvaccaro) [reply warehouse/scripts/dbt_artifacts/manifest.py:291]: yep it's a weird consequence of how they define the schemas; I actually asked someone about that, and he suggested opening up an issue to discuss

File: warehouse/scripts/dbt_artifacts/__init__.py (hunk 2 above)
Comment (atvaccaro) [reply warehouse/scripts/dbt_artifacts/__init__.py:256]: thanks I needed to remove these, not useful anymore

File: warehouse/scripts/dbt_artifacts/__init__.py (hunk 1 above)
Comment (atvaccaro) [reply warehouse/scripts/dbt_artifacts/__init__.py:92]: also need to remove ty

**Review by lauriemerrell — APPROVED** [review]

**Pull request merged**